cloudferrylib/os/actions/verify_vms.py

```diff
@@ -0,0 +1,85 @@
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+
+from cloudferrylib import utils
+from cloudferrylib.base import exception
+from cloudferrylib.base.action import action
+
+LOG = utils.get_log(__name__)
+
+
+class VerifyVms(action.Action):
+    """ Compare SRC and DST cloud resources. 'info' has DST resources. """
+    def run(self, **kwargs):
+        if not kwargs.get('info'):
+            raise exception.AbortMigrationError(
+                "No information from destination cloud."
+                "Something has been broken on early steps."
+            )
+        dst_info = kwargs['info']
+        search_opts = {'search_opts': kwargs.get('search_opts', {})}
+        search_opts.update(kwargs.get('search_opts_tenant', {}))
+        compute_resource = self.cloud.resources[utils.COMPUTE_RESOURCE]
+        src_info = compute_resource.read_info(**search_opts)
+        old_ids = set(dst_inst['meta']['old_id']
+                      for dst_inst in dst_info['instances'].values())
+        dst_cmp_info = {}
+        inst_cnt = 0
+        for dst_inst in dst_info['instances'].values():
+            old_id = dst_inst['meta']['old_id']
+            dst_cmp_info[old_id] = {}
+            dst_inst_ = dst_inst['instance']
+            dst_cmp_info[old_id].update(
+                {'name': dst_inst_['name']})
+            dst_cmp_info[old_id].update(
+                {'flav_details': dst_inst_['flav_details']})
+            dst_cmp_info[old_id].update(
+                {'key_name': dst_inst_['key_name']})
+            dst_cmp_info[old_id].update(
+                {'interfaces': dst_inst_['interfaces']})
+            dst_volumes = dst_inst['meta']['volume']
+            new_dst_volumes = []
+            for dst_vol in dst_volumes:
+                new_dst_volumes.append(dst_vol['volume'])
+            dst_cmp_info[old_id].update(
+                {'volumes': new_dst_volumes})
+            inst_cnt += 1
+        failed_vms = []
+        src_cmp_info = {}
+        for src_inst_id in src_info['instances']:
+            if src_inst_id not in old_ids:
+                failed_vms.append(src_inst_id)
+            else:
+                src_cmp_info[src_inst_id] = {}
+                src_inst_info = src_info['instances'][src_inst_id]['instance']
+                src_cmp_info[src_inst_id].update(
+                    {'name': src_inst_info['name']})
+                src_cmp_info[src_inst_id].update(
+                    {'flav_details': src_inst_info['flav_details']})
+                src_cmp_info[src_inst_id].update(
+                    {'key_name': src_inst_info['key_name']})
+                src_cmp_info[src_inst_id].update(
+                    {'interfaces': src_inst_info['interfaces']})
+                src_cmp_info[src_inst_id].update(
+                    {'volumes': src_inst_info['volumes']})
+        if failed_vms or (dst_cmp_info != src_cmp_info):
```

Comment:
In case when second condition is True (dst_cmp_info != src_cmp_info) we have warning message "Instances were not migrated:" and that's all. I think this check should display more useful information in this case.